tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -5252,25 +5272,29 @@ TEST_F(OpConverterTest, ConvertUnary) {
   // Add other unary ops to test.
   ops_to_test.push_back("Rsqrt");
   // Ok.
+  auto p = TestParamBase{{1, 1, 2, 3}, // input dims
+                         {},           // input partial dims
+                         {1, 1, 2, 3}, // expected output dims
+                        };
   for (const string& op_name : ops_to_test) {
-    Reset();
+    SCOPED_TRACE(p);
+    Reset(converter_precision, trt_mode);
     NodeDef node_def = get_unary_nodedef(op_name);
```

Comment:
Can we replace the two lambdas, get_unary_nodedef and get_unary_output, with a table of (opName, MakeNodeMethod, ComputeOutputMethod)?

Reply:
Something like this would be possible, let me know if you see a better option:

```c++
std::map<std::string,
         std::pair<std::function<Operation(Scope, ops::Placeholder)>,
                   std::function<float(float)>>>
    op_map;
op_map["Abs"] = std::make_pair(
    [](Scope s, ops::Placeholder p) { return ops::Abs(s, p).operation; },
    std::abs<float>);
````

Reply:
In the existing code, get_unary_nodedef creates the Scope and the Placeholder, so I think the first element in your pair doesn't need to take any parameter and can look like this
std::map<std::string, std::pair<std::function<operation(void)>, std::function<float(float)>>>;

We may want to use a template to define the first element in the pair:
template <class T>
Operation CreateOperation() {
   Scope s = Scope::NewRootScope();
    auto input = ops::Placeholder(s.WithOpName("input"), DT_FLOAT);
    return T(s.WithOpName("my_unary"), input);
}

#define ADD_OP(name, op, compute) op_map[name] = std::make_pair(CreateOperation<op>, compute<float>)

ADD_OP("Abs", ops::Abs, std::abs);

Would something like this work?

Reply:
Yes it works, and it is more concise.